### PR TITLE
Make generated code in syntactic sequence more stable

### PR DESCRIPTION
**Pull request by szarnekow**

### CODE REVIEW

File: org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java [hunk 1]
```diff
@@ -454,9 +454,33 @@ public AbstractElementAlias getAmbiguousSyntax() {
 				return ambiguousSyntax;
 			Nfa<ISynState> nfa = getAmbiguousNfa();
 			NfaToProduction prod = new NfaToProduction();
-			ambiguousSyntax = prod.nfaToGrammar(nfa, new GetGrammarElement(), new GrammarAliasFactory());
```
Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java]: we have no tests for this?

File: org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java [hunk 2]
```diff
@@ -454,9 +454,33 @@ public AbstractElementAlias getAmbiguousSyntax() {
...
+					grammarElement = nfa.getStop().getGrammarElement();
+					if (grammarElement == null) {
+						Iterator<ISynState> iter = nfa.getStart().getFollowers().iterator();
+						while (grammarElement == null && iter.hasNext()) {
+							grammarElement = iter.next().getGrammarElement();
+						}
+					}
+				}
+			}
+			Grammar grammar = GrammarUtil.getGrammar(grammarElement);
```
Comment (cdietrich) [inline org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java]: for debugging?

Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java]: The local variable? Yes.

File: org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java (hunk 1 above)
Comment (szarnekow) [reply org.eclipse.xtext/src/org/eclipse/xtext/serializer/analysis/SyntacticSequencerPDAProvider.java]: Not that I'm aware of. The instable generation manifested in a more involved grammar context.

**Pull request merged**